### ISSUE TITLE
Container shutdown hangs for 5 minutes, startup is also pretty sluggish

### ISSUE THREAD

**Issue by jonstelly** [docker/for-win#1788]
### Expected behavior
Builds should succeed, Containers should exit in a reasonable amount of time.

### Actual behavior
I get frequent timeouts when the layer containers are being shut down, this adds 5 minutes to most layers (Not all layers, but most, and I can't find a rhyme or reason to which ones have the timeout).  These delays often cause timeouts in my build scripts, so the builds fail.

I see this same timeout when doing a `docker run -it --rm ...` when I try to exit from the container, the command window will hang for 5 minutes until the container is terminated.

### Information
Diagnostic ID: EB7B58C5-173D-4889-8E5D-3D6D7D048FC0/2018-03-05_11-52-56
This is on a Surface Book 2 with 16GB of RAM.
Windows 10 Pro - 17083

I've tested it after a fresh reboot, tried it with nothing but a console window running.  Watching task manager, CPU and Disk performance seems reasonable.

Dockerfile:
```
FROM microsoft/aspnet
SHELL ["powershell", "-Command", "$ErrorActionPreference = 'Stop'; $ProgressPreference = 'SilentlyContinue';"]
RUN mkdir c:\site;
```

Interesting service log section (this is where the hang happens):

```
[11:34:27.171][WindowsDaemon  ][Info   ] Pending updates: false [module=libcontainerd namespace=moby container=1df30b9bc81191f08ebcc5b63a4007ef3b7e7a7a0fb5013a3775816fa4880895 process=init]
[11:34:27.171][WindowsDaemon  ][Info   ] debug: HCSShim::Container::Shutdown id=1df30b9bc81191f08ebcc5b63a4007ef3b7e7a7a0fb5013a3775816fa4880895
[11:34:27.196][WindowsDaemon  ][Info   ] debug: HCSShim::Container::WaitTimeout id=1df30b9bc81191f08ebcc5b63a4007ef3b7e7a7a0fb5013a3775816fa4880895
[11:39:00.031][WindowsDaemon  ][Info   ] svchost (5216,G,0) The beta feature EseDiskFlushConsistency is enabled in ESENT due to the beta site mode settings 0x800000.
[11:39:27.199][WindowsDaemon  ][Info   ] debug: failed to shutdown container, terminating it [error=container 1df30b9bc81191f08ebcc5b63a4007ef3b7e7a7a0fb5013a3775816fa4880895 encountered an error during WaitTimeout: hcsshim: timeout waiting for notification container=1df30b9bc81191f08ebcc5b63a4007ef3b7e7a7a0fb5013a3775816fa4880895 module=libcontainerd namespace=moby]
```

### Steps to reproduce the behavior

Using the above Dockerfile: `docker  build --no-cache . -t test`

**Comment by jonstelly** [comment]
Update: I've updated to the 17115 Insiders Build and this seems better, but still pretty slow.  I did another full Docker reset but the following step in my dockerfile:

`RUN mkdir c:\site\1; mkdir c:\site\2; mkdir c:\site\3; mkdir c:\site\4;`

still takes 30 seconds or more and I've seen a couple times where it took just over a minute.

As an editorial comment, I really wish Microsoft would support process isolation on Windows 10.  Hyperv isolation has so many drawbacks (resource allocation, start and stop times, look at how many issues here seem to be related to Windows 10 vs. Server 2016).  I'm pretty sure there are no technical reasons for it, only licensing/support reasons, and it's making the development experience for Docker on Windows vastly inferior to what it could be.
